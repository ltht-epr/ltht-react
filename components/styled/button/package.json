{
  "name": "@ltht-react/button",
  "version": "0.0.18",
  "description": "ltht-react atoms Button component.",
  "author": "LTHT",
  "homepage": "",
  "license": "MIT",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "directories": {
    "lib": "lib"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "clean": "rimraf lib",
    "build": "yarn build:lib",
    "watch": "tsc -w",
    "build:lib": "tsc",
    "type:check": "tsc --noEmit",
    "lint": "prettylint 'src/*.{ts,tsx}'"
  },
  "dependencies": {
    "@emotion/core": "^10.0.27",
    "@emotion/styled": "^10.0.27",
    "@ltht-react/styles": "^0.1.18",
    "@ltht-react/types": "^0.0.9",
    "@ltht-react/utils": "^0.1.18",
    "emotion": "^10.0.27",
    "react": "^16.12.0"
  },
  "gitHead": "4d364556bb11e294d86697c2dac66742f7aabc7f"
}
