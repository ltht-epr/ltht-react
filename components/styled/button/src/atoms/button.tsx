import { FC, ReactNode, HTMLAttributes } from 'react'
import styled from '@emotion/styled'
import { css, SerializedStyles } from '@emotion/react'
import { DESKTOP_MINIMUM_MEDIA_QUERY, BTN_COLOURS } from '@ltht-react/styles'

const setColors = (buttonStyle: string): SerializedStyles => {
  switch (buttonStyle) {
    case 'primary':
      return css`
        color: ${BTN_COLOURS.PRIMARY.TEXT};
        background-color: ${BTN_COLOURS.PRIMARY.VALUE};
        &:hover {
          background-color: ${BTN_COLOURS.PRIMARY.HOVER};
        }

        &:disabled {
          background-color: ${BTN_COLOURS.PRIMARY.DISABLED};
        }
      `
    case 'standard':
      return css`
        color: ${BTN_COLOURS.STANDARD.TEXT};
        background-color: ${BTN_COLOURS.STANDARD.VALUE};
        &:hover {
          background-color: ${BTN_COLOURS.STANDARD.HOVER};
        }

        &:disabled {
          background-color: ${BTN_COLOURS.STANDARD.DISABLED};
        }
      `
    case 'workflow':
      return css`
        color: ${BTN_COLOURS.WORKFLOW.TEXT};
        background-color: ${BTN_COLOURS.WORKFLOW.VALUE};
        &:hover {
          background-color: ${BTN_COLOURS.WORKFLOW.HOVER};
        }

        &:disabled {
          background-color: ${BTN_COLOURS.WORKFLOW.DISABLED};
        }
      `
    case 'danger':
      return css`
        color: ${BTN_COLOURS.DANGER.TEXT};
        background-color: ${BTN_COLOURS.DANGER.VALUE};
        &:hover {
          background-color: ${BTN_COLOURS.DANGER.HOVER};
        }

        &:disabled {
          background-color: ${BTN_COLOURS.DANGER.DISABLED};
        }
      `

    case 'clear':
      return css`
        color: ${BTN_COLOURS.CLEAR.TEXT};
        background-color: ${BTN_COLOURS.CLEAR.VALUE};
        &:hover {
          background-color: ${BTN_COLOURS.CLEAR.HOVER};
        }

        &:disabled {
          background-color: ${BTN_COLOURS.CLEAR.DISABLED};
        }
      `
    default:
      return css``
  }
}

const setIconMargins = (placement: IconPlacement): SerializedStyles => {
  switch (placement) {
    case 'left':
      return css`
        margin: 0 0.75rem 0 0;
      `
    case 'right':
      return css`
        margin: 0 0 0 0.75rem;
      `
    case 'center':
      return css`
        margin: 0;
      `
    default:
      return css``
  }
}

const StyledButton = styled.button<StyledProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  white-space: nowrap;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 0.8rem;
  font-weight: bold;
  line-height: 1.5;
  border-radius: 4px;
  width: 100%;

  &:hover:not([disabled]) {
    cursor: pointer;
  }

  &:disabled {
    opacity: 0.65;
    cursor: not-allowed;
  }

  ${DESKTOP_MINIMUM_MEDIA_QUERY} {
    font-size: 0.9rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: auto;
  }

  ${({ buttonStyle }): SerializedStyles => setColors(buttonStyle)}
`

const ButtonIcon = styled.div<ButtonIconProps>`
  display: flex;
  ${({ placement }): SerializedStyles => setIconMargins(placement)}
  svg {
    color: ${({ iconColour }) => iconColour};
  }
`

const ButtonText = styled.div``

const Button: FC<Props> = ({
  type,
  value,
  buttonStyle = 'primary',
  disabled = false,
  icon,
  iconPlacement = 'left',
  iconColour = 'white',
  ...rest
}) => (
  <StyledButton type={type} buttonStyle={buttonStyle} disabled={disabled} {...rest}>
    {icon && iconPlacement === 'left' && (
      <ButtonIcon placement={iconPlacement} iconColour={iconColour}>
        {icon}
      </ButtonIcon>
    )}
    {value && <ButtonText>{value}</ButtonText>}
    {icon && iconPlacement === 'right' && (
      <ButtonIcon placement={iconPlacement} iconColour={iconColour}>
        {icon}
      </ButtonIcon>
    )}
    {!value && icon && iconPlacement === 'center' && (
      <ButtonIcon placement={iconPlacement} iconColour={iconColour}>
        {icon}
      </ButtonIcon>
    )}
  </StyledButton>
)

type ButtonStyle = 'primary' | 'standard' | 'workflow' | 'danger' | 'clear'
type ButtonTypes = 'button' | 'submit' | 'reset'
type IconPlacement = 'left' | 'right' | 'center'

interface Props extends ButtonProps {
  buttonStyle?: ButtonStyle
}

interface StyledProps {
  buttonStyle: ButtonStyle
}

interface ButtonIconProps {
  placement: IconPlacement
  iconColour: string
}

export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
  type: ButtonTypes
  value?: string
  disabled?: boolean
  icon?: ReactNode
  iconPlacement?: IconPlacement
  iconColour?: string
}

export default Button
