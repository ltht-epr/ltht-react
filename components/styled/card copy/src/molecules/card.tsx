/** @jsx jsx */
import React from 'react'
import { css, jsx, SerializedStyles } from '@emotion/core'

import { CSS_RESET, TEXT_COLOURS, CARD_BACKGROUND_COLOUR, CARD_BACKGROUND_COLOUR_NO_DATA } from '@ltht-react/styles'
import CardEmpty from '../atoms/card-empty'

const styles = (noData: boolean): SerializedStyles => {
  return css`
  ${CSS_RESET}
  background: ${noData ? CARD_BACKGROUND_COLOUR_NO_DATA : CARD_BACKGROUND_COLOUR};
  color: ${noData ? TEXT_COLOURS.SECONDARY : TEXT_COLOURS.PRIMARY};
  margin-bottom: 1rem;
  border-radius: 4px;
  box-shadow: 
    0px 2px 1px -1px rgba(102, 102, 102, 0.1), 
    0px 1px 1px 0px rgba(102, 102, 102, 0.15), 
    0px 1px 3px 0px rgba(102, 102, 102,.6);
  -webkit-font-smoothing: antialiased;
`
}

const Card: React.FC<Props> = ({ children, noData = false }) => {
  return (
    <div css={styles(noData)}>
      {children}
      {noData && <CardEmpty />}
    </div>
  )
}

interface Props {
  noData?: boolean
}

export default Card
