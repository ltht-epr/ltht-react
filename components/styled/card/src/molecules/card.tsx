import React from 'react'
import classNames from 'classnames'
import styled from '@emotion/styled'
import { CSS_RESET, TEXT_COLOURS, CARD_BACKGROUND_COLOUR } from '@ltht-react/styles'
import Alert, { Props as AlertProps } from '../atoms/alert'
import Body, { Props as BodyProps } from '../atoms/body'
import Footer, { Props as FooterProps } from '../atoms/footer'
import Header, { Props as HeaderProps } from '../atoms/header'
import ListItem, { Props as ListItemProps } from '../atoms/list-item'
import List, { Props as ListProps } from '../atoms/list'
import Subtitle, { Props as SubtitleProps } from '../atoms/subtitle'
import Text, { Props as TextProps } from '../atoms/text'
import Title, { Props as TitleProps } from '../atoms/title'

interface CardComposition {
  Alert: React.FC<AlertProps>
  Body: React.FC<BodyProps>
  Footer: React.FC<FooterProps>
  Header: React.FC<HeaderProps>
  ListItem: React.FC<ListItemProps>
  List: React.FC<ListProps>
  Subtitle: React.FC<SubtitleProps>
  Text: React.FC<TextProps>
  Title: React.FC<TitleProps>
}

type TextAlignValues = 'left' | 'center' | 'right'

interface StyledProps {
  textAlign: TextAlignValues
}

interface Props {
  textAlign?: TextAlignValues
  classes?: string
}

const StyledCard = styled.div`
  ${CSS_RESET}
  position: relative;
  display: flex;
  flex-direction: column;
  text-align: ${(props: StyledProps): string => props.textAlign};
  background: ${CARD_BACKGROUND_COLOUR};
  color: ${TEXT_COLOURS.PRIMARY};
  margin-bottom: 1rem;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(102, 102, 102, 0.1), 0px 1px 1px 0px rgba(102, 102, 102, 0.15),
    0px 1px 3px 0px rgba(102, 102, 102, 0.6);
  -webkit-font-smoothing: antialiased;

  & > .card__alert + .card__list,
  > .card__alert + .card__body,
  > .card__list + .card__footer {
    border-top-width: 0;
  }
`

const Card: React.FC<Props> & CardComposition = ({ textAlign = 'left', classes, children }) => {
  return (
    <StyledCard textAlign={textAlign} className={classNames('card', classes)}>
      {children}
    </StyledCard>
  )
}

Card.Alert = Alert
Card.Body = Body
Card.Footer = Footer
Card.Header = Header
Card.ListItem = ListItem
Card.List = List
Card.Subtitle = Subtitle
Card.Text = Text
Card.Title = Title

export default Card
