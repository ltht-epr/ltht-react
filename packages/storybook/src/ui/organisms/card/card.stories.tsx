import { Story } from '@storybook/react'

import Card from '@ltht-react/card'
import AllergySummary from '@ltht-react/allergy-summary'
import { InfoSummary, WarningSummary, ErrorSummary, MissingDataSummary } from '@ltht-react/type-summary'
import allergies from '../../../clinical/organisms/allergies/allergies.fixtures'

const clickHandler = (): void => {
  // eslint-disable-next-line no-console
  console.log('Handler clicked')
}

export const Basic: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.Body>
      <Card.Subtitle>Subtitle</Card.Subtitle>
      <Card.Text>Body Text</Card.Text>
    </Card.Body>
  </Card>
)

export const WithFooter: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.Body>
      <Card.Subtitle>Subtitle</Card.Subtitle>
      <Card.Text>Body Text</Card.Text>
    </Card.Body>
    <Card.Footer>Footer</Card.Footer>
  </Card>
)

export const WithList: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.List>
      <Card.ListItem>Item 1</Card.ListItem>
      <Card.ListItem>Item 2</Card.ListItem>
    </Card.List>
  </Card>
)

export const ClickableList: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Header</Card.Title>
    </Card.Header>
    <Card.List>
      <Card.ListItem onClick={clickHandler}>
        <div>Item 1</div>
      </Card.ListItem>
      <Card.ListItem onClick={clickHandler}>
        <div>Item 2</div>
      </Card.ListItem>
    </Card.List>
  </Card>
)

export const InfoSummaryStory: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <InfoSummary clickHandler={clickHandler} />
    <Card.List>
      {allergies.map((allergy) => (
        <Card.ListItem key={allergy.id}>
          <AllergySummary allergy={allergy} />
        </Card.ListItem>
      ))}
    </Card.List>
    <InfoSummary clickHandler={clickHandler} />
    <Card.List>
      <Card.ListItem>Missing 10</Card.ListItem>
    </Card.List>
    <WarningSummary clickHandler={clickHandler} />
    <Card.List>
      <Card.ListItem>Missing 10</Card.ListItem>
      <Card.ListItem>Missing 10</Card.ListItem>
    </Card.List>
  </Card>
)
InfoSummaryStory.storyName = 'Info Summary'

export const WarningSummaryStory: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <WarningSummary clickHandler={clickHandler} />
    <Card.List>
      {allergies.map((allergy) => (
        <Card.ListItem key={allergy.id}>
          <AllergySummary allergy={allergy} />
        </Card.ListItem>
      ))}
    </Card.List>
  </Card>
)
WarningSummaryStory.storyName = 'Warning Summary'

export const ErrorSummaryStory: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <ErrorSummary clickHandler={clickHandler} />
    <Card.List>
      <Card.ListItem>No Data</Card.ListItem>
    </Card.List>
  </Card>
)
ErrorSummaryStory.storyName = 'Error Summary'

export const MissingDataSummaryStory: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
    <MissingDataSummary
      type="allergies"
      noData={['Item 1', 'Item 2']}
      missingData={['Item 1', 'Item 2']}
      partialData={['Item 1', 'Item 2']}
      completeData={['Item 1', 'Item 2']}
    />
  </Card>
)
MissingDataSummaryStory.storyName = 'Missing Data Summary'

export const NoData: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergies</Card.Title>
    </Card.Header>
  </Card>
)

export default { title: 'UI/Organisms/Card' }
