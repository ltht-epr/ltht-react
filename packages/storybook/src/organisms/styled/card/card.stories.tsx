import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import readme from '@ltht-react/card/README.md'
import { ListItem } from '@ltht-react/list'
import { MissingDataSummary, MissingDataWarning } from '@ltht-react/summary'
import { Card, CardHeader, CardBody, CardFooter, CardList } from '@ltht-react/card'

const stories = storiesOf('Organisms - Styled|Card', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Basic', () => (
  <Card>
    <CardHeader>
      <h3>Header</h3>
    </CardHeader>
    <CardBody>Body</CardBody>
  </Card>
))

stories.addWithJSX('Footer', () => (
  <Card>
    <CardHeader>
      <h3>Header</h3>
    </CardHeader>
    <CardBody>Body</CardBody>
    <CardFooter>Footer</CardFooter>
  </Card>
))

stories.addWithJSX('List', () => (
  <div>
    <Card>
      <CardHeader>
        <h3>Header</h3>
      </CardHeader>
      <CardBody>
        <CardList>
          <ListItem>Item 1</ListItem>
          <ListItem>Item 2</ListItem>
        </CardList>
      </CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
  </div>
))

stories.addWithJSX('List (Clickable)', () => (
  <div>
    <Card>
      <CardHeader>
        <h3>Header</h3>
      </CardHeader>
      <CardBody>
        <CardList clickable>
          <ListItem>Item 1</ListItem>
          <ListItem>Item 2</ListItem>
        </CardList>
      </CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
  </div>
))

stories.addWithJSX('Missing Data Warning', () => (
  <div>
    <Card>
      <CardHeader>
        <h3>Header</h3>
      </CardHeader>
      <CardBody>
        <CardList clickable>
          <MissingDataWarning />
          <ListItem>Item 1</ListItem>
          <ListItem>Item 2</ListItem>
        </CardList>
      </CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
  </div>
))

stories.addWithJSX('Missing Data Summary', () => (
  <div>
    <Card>
      <CardHeader position="center">
        <h3>Header</h3>
      </CardHeader>
      <CardBody>
        <CardList clickable>
          <MissingDataSummary
            noData={['Item 1', 'Item 2']}
            partialData={['Item 3', 'Item 4']}
            allData={['Item 5', 'Item 6']}
          />
        </CardList>
      </CardBody>
      <CardFooter>Footer</CardFooter>
    </Card>
  </div>
))

stories.addWithJSX('No Data', () => (
  <div>
    <Card noData>
      <CardHeader>
        <h3>Header</h3>
      </CardHeader>
    </Card>
  </div>
))
