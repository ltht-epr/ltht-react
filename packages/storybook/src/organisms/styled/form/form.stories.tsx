/** @jsx jsx */
import React from 'react'
import { css, jsx } from '@emotion/core'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import readme from '@ltht-react/form/README.md'
import { Card, CardHeader, CardBody } from '@ltht-react/card'
import { TEXT_COLOURS, DESKTOP_MEDIA_QUERY } from '@ltht-react/styles'
import { Form, FormCheck } from '@ltht-react/form'
import { RadioButton } from '@ltht-react/input'
import { PrimaryButton } from '@ltht-react/button'

const styles = {
  base: css`
    margin-top: 5px;
    margin-left: 5px;
    ${DESKTOP_MEDIA_QUERY} {
      display: flex;
      align-items: center;
      justify-content: center;
      margin-left: 0;
    }
  `,
  text: css`
    color: ${TEXT_COLOURS.SECONDARY};
    padding-bottom: 0.3rem;

    ${DESKTOP_MEDIA_QUERY} {
      padding-bottom: 0.5rem;
    }
  `,
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const stories = storiesOf('Organisms - Styled|Form', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Radio', () => {
  const [reason, setReason] = React.useState('')
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => setReason(e.currentTarget.value)
  const handleSubmit = (evt: React.FormEvent<HTMLFormElement>): void => {
    evt.preventDefault()
  }

  return (
    <div css={styles.base}>
      <Card>
        <CardHeader>
          <h3>Form example</h3>
        </CardHeader>
        <CardBody>
          <Form submitHandler={handleSubmit}>
            <p css={styles.text}>Please select the option</p>
            <FormCheck>
              <RadioButton id="1" value="1" label="Option 1" checked={reason === '1'} changeHandler={handleChange} />
            </FormCheck>
            <FormCheck>
              <RadioButton id="2" value="2" label="Option 2" checked={reason === '2'} changeHandler={handleChange} />
            </FormCheck>
            <PrimaryButton type="submit" value="Submit" disabled={reason === ''} />
          </Form>
        </CardBody>
      </Card>
    </div>
  )
})
