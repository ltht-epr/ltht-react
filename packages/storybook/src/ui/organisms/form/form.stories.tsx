import { useState, ChangeEvent, FormEvent } from 'react'
import { Story } from '@storybook/react'
import styled from '@emotion/styled'

import Card from '@ltht-react/card'
import { TEXT_COLOURS, DESKTOP_MINIMUM_MEDIA_QUERY } from '@ltht-react/styles'
import { Form, FormCheck } from '@ltht-react/form'
import { RadioButton } from '@ltht-react/input'
import { PrimaryButton } from '@ltht-react/button'

const StyledForm = styled.div`
  margin-top: 5px;
  margin-left: 5px;

  ${DESKTOP_MINIMUM_MEDIA_QUERY} {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-left: 0;
  }
`

const StyledText = styled.p`
  color: ${TEXT_COLOURS.SECONDARY};
  margin: 0;
  padding-bottom: 0.3rem;

  ${DESKTOP_MINIMUM_MEDIA_QUERY} {
    padding-bottom: 0.5rem;
  }
`

export const Radio: Story = () => {
  const [reason, setReason] = useState('')
  const handleChange = (e: ChangeEvent<HTMLInputElement>): void => setReason(e.currentTarget.value)
  const handleSubmit = (evt: FormEvent<HTMLFormElement>): void => {
    evt.preventDefault()
  }

  return (
    <StyledForm>
      <Card>
        <Card.Header>
          <Card.Title>Form Example</Card.Title>
        </Card.Header>
        <Card.Body>
          <Form submitHandler={handleSubmit}>
            <StyledText>Please select the option</StyledText>
            <FormCheck>
              <RadioButton id="1" value="1" label="Option 1" checked={reason === '1'} changeHandler={handleChange} />
            </FormCheck>
            <FormCheck>
              <RadioButton id="2" value="2" label="Option 2" checked={reason === '2'} changeHandler={handleChange} />
            </FormCheck>
            <PrimaryButton type="submit" value="Submit" disabled={reason === ''} />
          </Form>
        </Card.Body>
      </Card>
    </StyledForm>
  )
}

export default { title: 'UI/Organisms/Form' }
