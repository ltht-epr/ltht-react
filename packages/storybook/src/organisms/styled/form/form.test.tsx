import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import { Card, CardBody, CardHeader } from '@ltht-react/card'
import { FormCheck, Form } from '@ltht-react/form'
import { RadioButton } from '@ltht-react/input'
import { PrimaryButton } from '@ltht-react/button'

const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
  e.preventDefault()
}
const handleSubmit = (evt: React.FormEvent<HTMLFormElement>): void => {
  evt.preventDefault()
}

const components = (
  <Card>
    <CardHeader>
      <h3>Form example</h3>
    </CardHeader>
    <CardBody>
      <Form submitHandler={handleSubmit}>
        <FormCheck>
          <RadioButton id="1" value="1" label="Option 1" checked changeHandler={handleChange} />
        </FormCheck>
        <FormCheck>
          <RadioButton id="2" value="2" label="Option 2" checked={false} changeHandler={handleChange} />
        </FormCheck>
        <PrimaryButton type="submit" value="Submit" disabled />
      </Form>
    </CardBody>
  </Card>
)

describe('Form', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(components, div)
  })
  it('matches snapshot', () => {
    expect(mount(components)).toMatchSnapshot('wrapper mount')
  })
})
