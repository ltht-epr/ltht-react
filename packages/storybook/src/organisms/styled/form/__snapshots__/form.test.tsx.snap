// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches snapshot: wrapper mount 1`] = `
.emotion-8 {
  color: #231f20;
  font-family: 'SF Pro Text','SF Pro Icons','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.8rem;
  background: #FFFFFF;
  color: #231f20;
  margin-bottom: 1rem;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(102,102,102,0.1), 0px 1px 1px 0px rgba(102,102,102,0.15), 0px 1px 3px 0px rgba(102,102,102,.6);
  -webkit-font-smoothing: antialiased;
}

.emotion-8 *,
.emotion-8 ::before,
.emotion-8 ::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.emotion-8 * {
  margin: 0;
  padding: 0;
}

.emotion-8 h3 {
  font-size: 1rem;
}

@media only screen and (min-width:768px) {
  .emotion-8 {
    font-size: 0.9rem;
  }

  .emotion-8 h3 {
    font-size: 1.1rem;
  }
}

@media only screen and (min-width:1024px) {
  .emotion-8 {
    font-size: 1rem;
  }

  .emotion-8 h3 {
    font-size: 1.2rem;
  }
}

@media only screen and (min-width:1200px) {
  .emotion-8 {
    font-size: 1.1rem;
  }

  .emotion-8 h3 {
    font-size: 1.3rem;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem !important;
}

.emotion-7 {
  padding: 0 0.5rem 0 0.5rem !important;
}

.emotion-7 > dl:first-of-type {
  margin-top: 0 !important;
}

.emotion-6 {
  margin: 0rem;
}

.emotion-2 {
  display: block;
  position: relative;
  margin-bottom: 0.5rem;
}

.emotion-1 {
  padding-left: 1.25rem;
  background-color: #ddefc7;
}

.emotion-1 input {
  position: absolute;
  margin-top: 0.15rem;
  margin-left: -1.25rem;
}

.emotion-1 label {
  display: inline-block;
  margin-bottom: 0;
}

.emotion-3 {
  padding-left: 1.25rem;
}

.emotion-3 input {
  position: absolute;
  margin-top: 0.15rem;
  margin-left: -1.25rem;
}

.emotion-3 label {
  display: inline-block;
  margin-bottom: 0;
}

.emotion-5 {
  display: block;
  text-align: center;
  white-space: nowrap;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 4px;
  width: 100%;
  color: #FFFFFF;
  background-color: #003087;
}

.emotion-5:hover:not([disabled]) {
  cursor: pointer;
}

.emotion-5:disabled {
  opacity: 0.65;
}

@media only screen and (min-width:1024px) {
  .emotion-5 {
    display: inline-block;
    vertical-align: middle;
    width: auto;
  }
}

.emotion-5:hover {
  background-color: #002a6e;
}

.emotion-5:disabled {
  background-color: #406aae;
}

<Card>
  <div
    className="emotion-8"
  >
    <CardHeader>
      <div
        className="emotion-0"
        onClick={[Function]}
        title=""
      >
        <h3>
          Form example
        </h3>
      </div>
    </CardHeader>
    <CardBody>
      <div
        className="emotion-7"
      >
        <Form
          submitHandler={[Function]}
        >
          <form
            className="emotion-6"
            onSubmit={[Function]}
          >
            <FormCheck>
              <div
                className="form-check emotion-2"
              >
                <RadioButton
                  changeHandler={[Function]}
                  checked={true}
                  id="1"
                  label="Option 1"
                  value="1"
                >
                  <div
                    className="emotion-1"
                  >
                    <input
                      checked={true}
                      id="1"
                      onChange={[Function]}
                      type="radio"
                      value="1"
                    />
                    <label
                      htmlFor="1"
                    >
                      Option 1
                    </label>
                  </div>
                </RadioButton>
              </div>
            </FormCheck>
            <FormCheck>
              <div
                className="form-check emotion-2"
              >
                <RadioButton
                  changeHandler={[Function]}
                  checked={false}
                  id="2"
                  label="Option 2"
                  value="2"
                >
                  <div
                    className="emotion-3"
                  >
                    <input
                      checked={false}
                      id="2"
                      onChange={[Function]}
                      type="radio"
                      value="2"
                    />
                    <label
                      htmlFor="2"
                    >
                      Option 2
                    </label>
                  </div>
                </RadioButton>
              </div>
            </FormCheck>
            <PrimaryButton
              disabled={true}
              type="submit"
              value="Submit"
            >
              <Button
                buttonStyle="primary"
                disabled={true}
                type="submit"
                value="Submit"
              >
                <button
                  className="emotion-5"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </Button>
            </PrimaryButton>
          </form>
        </Form>
      </div>
    </CardBody>
  </div>
</Card>
`;
